lib/samples/display_clusters/display_clusters.dart

```diff
@@ -90,22 +106,111 @@ class _DisplayClustersState extends State<DisplayClusters>
     _mapViewController.arcGISMap = _map;
     // Load the map.
     await _map.load();
-    // Set the ready state variable to true to enable the sample UI.
-    setState(() => _ready = true);
-  }
-
-  void toggleFeatureClustering() {
-    // Check if the map has operational layers and the first layer is a feature layer.
+    // Get the power plant feature layer once the map has finished loading.
     if (_map.operationalLayers.isNotEmpty &&
         _map.operationalLayers.first is FeatureLayer) {
-      // Get the first layer as a feature layer.
-      final featureLayer = _map.operationalLayers.first as FeatureLayer;
-      // Check if the feature layer has feature reduction.
-      if (featureLayer.featureReduction != null) {
-        // Toggle the feature reduction.
-        featureLayer.featureReduction!.enabled =
-            !featureLayer.featureReduction!.enabled;
+      // Get the first layer from the web map a feature layer.
+      _featureLayer = _map.operationalLayers.first as FeatureLayer;
+      if (_featureLayer.featureReduction != null) {
+        // Set the ready state variable to true to enable the sample UI.
+        // Set the feature reduction flag to the current state of the feature
+        setState(() {
+          _ready = true;
+          _featureReductionEnabled = _featureLayer.featureReduction!.enabled;
+        });
+      } else {
+        showWarningDialog(
+          'Feature layer does not have feature reduction enabled.',
+        );
       }
+    } else {
+      showWarningDialog('Unable to access a feature layer on the web map.');
+    }
+  }
+
+  void onTap(Offset localPosition) async {
+    // Clear any existing selected features.
+    _featureLayer.clearSelection();
+    // Perform an identify result on the map view controller, using the feature layer and tapped location.
+    final identifyLayerResult = await _mapViewController.identifyLayer(
+      _featureLayer,
+      screenPoint: localPosition,
+      tolerance: 12.0,
+    );
+    // Get the aggregate geoelements from the identify result.
+    final aggregateGeoElements =
+        identifyLayerResult.geoElements.whereType<AggregateGeoElement>();
+    if (aggregateGeoElements.isEmpty) return;
+    // Select the first aggregate geoelement.
+    final aggregateGeoElement = aggregateGeoElements.first;
+    aggregateGeoElement.isSelected = true;
+
+    // Get the list of geoelements associated with the aggregate geoelement.
+    final geoElements = await aggregateGeoElement.getGeoElements();
+    // Display a dialog with information about the geoelements.
+    showResultsDialog(geoElements);
+  }
+
+  void showResultsDialog(List<GeoElement> geoElements) {
+    // Create a dialog that lists the count and names of the provided list of geoelements.
+    showDialog(
+      context: context,
+      builder: (context) {
+        return AlertDialog(
+          scrollable: true,
```

Comment:
Ah - no we don't, I've removed the `scrollable: true`. I was experimenting after seeing that property and forgot to remove. It does work, to scroll the entire dialog but then actually it's the list of geoelements I want to scroll, rather than all the content (title too etc). I had to add `mainAxisSize: MainAxisSize.min,` as `scrollable: true` seems to have been constraining the height.